lusSTR/wrappers/filter.py

```diff
@@ -431,41 +513,39 @@ def main(
         raise ValueError(f"unknown data type '{data_type}'")
     if output_type not in ("efm", "strmix", "mpsproto"):
         raise ValueError(f"unknown output type '{output_type}'")
-    full_df = pd.read_csv(input, sep="\t")
     if output_dir is None:
         raise ValueError("No output specified using --out.")
-    else:
-        outpath = output_dir
-    if custom:
-        seq_col = "Custom_Range_Sequence"
-        brack_col = "Custom_Bracketed_Notation"
-    else:
-        seq_col = "UAS_Output_Sequence" if strand == "uas" else "Forward_Strand_Sequence"
-        brack_col = (
-            "UAS_Output_Bracketed_Notation"
-            if strand == "uas"
-            else "Forward_Strand_Bracketed_Notation"
+    if sex:
+        outpath_sex = f"{output_dir}/ystrs/"
+        input_name_sex = f"{os.path.splitext(input)[0]}_sexloci"
+        process_input(
+            input_name_sex,
+            outpath_sex,
+            profile_type,
+            data_type,
+            output_type,
+            strand=strand,
+            nofiltering=nofilters,
+            separate=separate,
+            custom=custom,
+            sex=sex,
+            info=info,
         )
-    if nofilters:
-        full_df["allele_type"] = "real_allele"
-        marker_plots(full_df, outpath)
-        if output_type == "efm" or output_type == "mpsproto":
-            EFM_output(full_df, outpath, profile_type, data_type, brack_col, separate)
-        else:
-            STRmix_output(full_df, outpath, profile_type, data_type, seq_col)
-    else:
-        dict_loc = {k: v for k, v in full_df.groupby(["SampleID", "Locus"])}
-        final_df, flags_df = process_strs(dict_loc, data_type, seq_col, brack_col)
-        marker_plots(final_df, outpath)
-        if output_type == "efm" or output_type == "mpsproto":
-            EFM_output(final_df, outpath, profile_type, data_type, brack_col, separate)
-        else:
-            STRmix_output(final_df, outpath, profile_type, data_type, seq_col)
-        if info:
-            name = os.path.basename(outpath)
-            final_df.to_csv(f"{outpath}/{name}_sequence_info.csv", index=False)
-            if not flags_df.empty:
-                flags_df.to_csv(f"{outpath}/{name}_Flagged_Loci.csv", index=False)
+    input_name = os.path.splitext(input)[0]
+    outpath = output_dir
+    process_input(
+        input_name,
+        outpath,
+        profile_type,
+        data_type,
+        output_type,
+        strand=strand,
+        nofiltering=nofilters,
+        separate=separate,
+        custom=custom,
+        sex=sex,
+        info=info,
+    )
```

Comment:
This is called whether `sex` is true or false? To make sure I understand, enabling `sex` includes processing of the Y-STRs in addition to the autosomals?

Reply:
Correct